src/plugins/chatview/views/UserDetailsModal.js

```diff
@@ -0,0 +1,97 @@
+import log from "@converse/headless/log";
```

Comment:
Modals are being moved to the `src/modals` folder. AFAIK, there's nothing about the `UserDetailsModal` that makes it specific to this plugin, and I think it should therefore also go there.

This also brings up the topic of which files to keep inside plugins and which files to make more "global" in order to facilitate code reuse, and this applies to modals but also to components (and probably other files as well).

I'm concerned that if we put relatively general purpose modals and components inside plugin folders, that we're creating too tight coupling between them and the plugins. I'm also concerned that if we have a "global" `modals` or `components` folders, and also the same folders inside the plugin folders, that we make discoverability harder.

Reply:
Moreover, if we are working on a plugin-based architecture, maybe we should avoid as much as possible, if not completely, anything more global than a plugin. If the plugin architecture is strong enough, and I think it is, we should probably put any "global" code in a plugin of its own. E.g.
- `converse-modals`
- `converse-utils`
- etc

Reply:
>  Why not have a converse-core or converse-global plugin for global elements such as modals?

The purpose of plugins is to make it possible to add and/or remove functionality to Converse. In other words, for something to be a plugin, it must make sense to be able to remove that plugin from your build and still have something functional (for a particular use-case of course).

Seen in this way, it doesn't make sense to put code that is so core to Converse that it won't be able to function without it in a plugin.

That said, we can probably have a `core`, `global` or `shared` folder which contains those files that are too central to belong to any one plugin.

> Or for modals, even have its own plugin, converse-modals

We do have a `converse-modal` plugin, but it's more concerned with making it possible to create modals, and I didn't consider it as a container for all "global" modals. But now that you mention it, I guess that could work yeah.

So we can put plugin-specific modals in the plugin itself, and more general modals go into the `modal` plugin folder.

I still think we might run into situations where it's not 100% clear where something should go or where it's confusing to people that a particular file is in one place and not somewhere else, but I think generally the layout is logical.

> Moreover, if we are working on a plugin-based architecture, maybe we should avoid as much as possible, if not completely, anything more global than a plugin

I think we can get quite far with plugins, but I'd be wary of abusing the notion of a plugin in order to maintain some kind of file structure, and as mentioned above, it doesn't make semantic sense to me to put code that is so central that Converse can't ever function without it, inside a plugin.

For the headless build, that's mainly the code in `converse-core.js`, for the UI layer, we might actually be able to put everything into plugins, we'll have to see.